libosmscout-map-qt/src/osmscout/MapPainterQt.cpp

```diff
@@ -480,10 +484,41 @@ namespace osmscout {
 
     QTransform tran;
 
+    // precompute character widths
+    std::vector<qreal> characterWidths(string.size());
+    for (int i=0; i<string.size(); i++) {
+      characterWidths[i]=metrics.width(string[i]);
+    }
+
     while (offset<p.length()) {
```

Comment:
I move this logic to `SimplifiedPath::TestAngleVariance`. It may be rewrited when this method will drop too many labels...